package/kernel/linux/modules/multimedia.mk

```diff
@@ -0,0 +1,756 @@
+ The STV06XX USB Camera Driver (stv06xx) kernel module
+endef
+
+$(eval $(call KernelPackage,video-gspca-stv06xx))
+
+
+define KernelPackage/video-gspca-gl860
+  TITLE:=gl860 webcam support
+  KCONFIG:=CONFIG_USB_GL860
+  FILES:=$(LINUX_DIR)/drivers/media/usb/gspca/gl860/gspca_gl860.ko
+  AUTOLOAD:=$(call AutoProbe,gspca_gl860)
+  $(call AddDepends/camera-gspca)
+endef
+
+define KernelPackage/video-gspca-gl860/description
+ The GL860 USB Camera Driver (gl860) kernel module
+endef
+
+$(eval $(call KernelPackage,video-gspca-gl860))
+
+
+define KernelPackage/video-gspca-jeilinj
+  TITLE:=jeilinj webcam support
+  KCONFIG:=CONFIG_USB_GSPCA_JEILINJ
+  FILES:=$(LINUX_DIR)/drivers/media/usb/gspca/gspca_jeilinj.ko
+  AUTOLOAD:=$(call AutoProbe,gspca_jeilinj)
+  $(call AddDepends/camera-gspca)
+endef
+
+define KernelPackage/video-gspca-jeilinj/description
+ The JEILINJ USB Camera Driver (jeilinj) kernel module
+endef
+
+$(eval $(call KernelPackage,video-gspca-jeilinj))
+
+
+define KernelPackage/video-gspca-konica
+  TITLE:=konica webcam support
+  KCONFIG:=CONFIG_USB_GSPCA_KONICA
+  FILES:=$(LINUX_DIR)/drivers/media/usb/gspca/gspca_konica.ko
+  AUTOLOAD:=$(call AutoProbe,gspca_konica)
+  $(call AddDepends/camera-gspca)
+endef
+
+define KernelPackage/video-gspca-konica/description
+ The Konica USB Camera Driver (konica) kernel module
+endef
+
+$(eval $(call KernelPackage,video-gspca-konica))
+
+# DVB & tuners
+
+define KernelPackage/dvb-usb-v2
+  SUBMENU:=$(MMEDIA_MENU)
+  TITLE:=USB DVB support
+  KCONFIG:= \
+	CONFIG_MEDIA_DIGITAL_TV_SUPPORT=y \
+	CONFIG_MEDIA_ANALOG_TV_SUPPORT=y \
+	CONFIG_DVB_CORE=m \
+	CONFIG_DVB_USB_V2=m
+  DEPENDS:=+kmod-i2c-mux +kmod-usb-core +kmod-multimedia
+  FILES:= \
+	$(LINUX_DIR)/drivers/media/dvb-core/dvb-core.ko \
+	$(LINUX_DIR)/drivers/media/usb/dvb-usb-v2/dvb_usb_v2.ko
+  AUTOLOAD:=$(call AutoLoad,81,dvb-core dvb_usb_v2)
+endef
+define KernelPackage/dvb-usb-v2/description
+  Kernel module for USB DVB support
+endef
+$(eval $(call KernelPackage,dvb-usb-v2))
+
+define KernelPackage/dvb-usb-rtl28xx
+  SUBMENU:=$(MMEDIA_MENU)
+  TITLE:=DVB USB Realtek RTL28xx
+  KCONFIG:= \
+	CONFIG_DVB_USB_RTL28XXU=m \
+	CONFIG_DVB_RTL2830=m \
+	CONFIG_DVB_RTL2832=m
+  DEPENDS:=+kmod-regmap-i2c +kmod-dvb-usb-v2
+  FILES:= \
+	$(LINUX_DIR)/drivers/media/usb/dvb-usb-v2/dvb-usb-rtl28xxu.ko \
+	$(LINUX_DIR)/drivers/media/dvb-frontends/rtl2830.ko \
+	$(LINUX_DIR)/drivers/media/dvb-frontends/rtl2832.ko
+  AUTOLOAD:=$(call AutoProbe,82,dvb-usb-rtl28xxu rtl2830 rtl2832)
+endef
+define KernelPackage/dvb-usb-rtl28xx/description
+  Kernel modules for RTL28xxU USB DVB demod and RTL2830 & RTL2832 DVB-T frontends
+endef
+$(eval $(call KernelPackage,dvb-usb-rtl28xx))
+
+define KernelPackage/dvb-tuner-r820t
+  SUBMENU:=$(MMEDIA_MENU)
+  TITLE:=DVB Tuner Rafael Micro R820T
+  KCONFIG:=CONFIG_MEDIA_TUNER_R820T=m
+  DEPENDS:=+kmod-i2c-core +kmod-multimedia
+  FILES:=$(LINUX_DIR)/drivers/media/tuners/r820t.ko
+  AUTOLOAD:=$(call AutoProbe,82,r820t)
```

Comment:
This seems to result in a `/etc/modules.d/r820t` with content `82`… which seems wrong. Seems like you can just drop the `82,` part. [also in other `AutoProbe` calls, that's just an example]

Reply:
I'll remove the "82".
Initially it was AutoLoad but I noticed that many other modules for USB devices were using AutoProbe. I asked for an explanation in the [forum ](https://forum.openwrt.org/t/what-does-call-autoprobe-do-in-kernel-module-packages/142663)but nobody answered. I still don't understand how it works.

Reply:
[`AutoProbe` is defined in `include/kernel.mk`](https://github.com/openwrt/openwrt/blob/master/include/kernel.mk#L272-L283) as an alias for the just before defined `AutoLoad` to avoid setting its first parameter *priority* (which in turn defaults it to `0`). So `AutoProbe,foo`, `AutoLoad,,foo` and `AutoLoad,0,foo` are exchangeable for a module `foo`.